File: Provenance/Game Library/PVRecentGame.m
//
//  PVRecentGame.m
//  Provenance
//
//  Created by James Addyman on 29/09/2015.
//  Copyright © 2015 James Addyman. All rights reserved.
//

#import "PVRecentGame.h"

@implementation PVRecentGame

- (instancetype)initWithGame:(PVGame *)game
{
    if ((self = [super init]))
    {
        self.game = game;
        self.lastPlayedDate = [NSDate date];
    }

    return self;
}

@end

File: Provenance/Game Library/Models/PVRecentGame.swift
//
//  PVRecentGame.swift
//  Provenance
//
//  Created by Joe Mattiello on 10/02/2018.
//  Copyright (c) 2018 JamSoft. All rights reserved.
//

import Foundation
import RealmSwift

@objcMembers public class PVRecentGame: Object, PVLibraryEntry {

    dynamic var game: PVGame!
    dynamic var lastPlayedDate: Date = Date()
	dynamic var core: PVCore?

    override public static func indexedProperties() -> [String] {
        return ["lastPlayedDate"]
    }

	public convenience init(withGame game: PVGame, core: PVCore? = nil) {
        self.init()
        self.game = game
		self.core = core
    }
}
